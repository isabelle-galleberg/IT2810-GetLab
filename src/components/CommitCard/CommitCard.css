.commitCard {
  background-color: #f2f1f0;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  border: 1px solid #3a73c5;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  gap: 5px;
  padding: 5px;
  height: 100%;
  width: clamp(200px, calc(75vw), 400px);
}

.title {
  font-size: 12px;
  font-weight: bold;
}

.commitInfo {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 5px;
}

.committedAt,
.author {
  font-size: 10px;
}

.committedAt {
  color: grey;
}

.profileImage {
  border-radius: 50px;
  height: 30px;
  width: 30px;
}

@media screen and (max-width: 500px) {
  .title {
    font-size: 10px;
  }
  .committedAt,
  .author {
    font-size: 8px;
  }
}
