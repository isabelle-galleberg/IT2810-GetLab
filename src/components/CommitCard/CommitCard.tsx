import { CommitCardProps } from "../../types/propTypes";
import "./CommitCard.css";

export default function CommitCard({
  title,
  committedAt,
  author,
}: CommitCardProps) {
  return (
    <div className="commitCard">
      <div className="title">{title}</div>
      <div className="commitInfo">
        <div className="author">{author}</div>
        <div className="committedAt">{`commited on ${committedAt}`}</div>
      </div>
    </div>
  );
}
