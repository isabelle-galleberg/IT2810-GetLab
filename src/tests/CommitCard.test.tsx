import renderer from "react-test-renderer";
import CommitCard from "../components/CommitCard/CommitCard";

describe("<CommitCard />", () => {
  it("should render a commit card with given values", () => {
    const elem = renderer
      .create(
        <CommitCard
          title="My first commit"
          committedAt="10.08.2022"
          author="me"
        />
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
