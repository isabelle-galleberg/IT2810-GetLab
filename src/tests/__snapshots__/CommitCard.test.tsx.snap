// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommitCard /> should render a commit card with given values 1`] = `
<div
  className="commitCard"
>
  <div
    className="title"
  >
    My first commit
  </div>
  <div
    className="commitInfo"
  >
    <div
      className="author"
    >
      me
    </div>
    <div
      className="committedAt"
    >
      commited on 10.08.2022
    </div>
  </div>
</div>
`;
